aten/src/ATen/TensorSubclassLikeUtils.h

```diff
@@ -28,8 +28,7 @@ constexpr auto kFunctorchWrappedTensors = DispatchKeySet({
 
 constexpr auto kTensorSubclassLike = kFunctorchWrappedTensors | DispatchKeySet({
     DispatchKey::Batched,
-    DispatchKey::SparseCPU,
-    DispatchKey::SparseCUDA,
+    DispatchKey::Sparse,
```

Comment:
This change is actually required. Suppose you wrote code like this:
```
auto sparse_cpu_ks = DispatchKeySet(DispatchKey::SparseCPU);
return (my_ks & sparse_cpu_ks).empty()
```

Previously, this would only return true if the keyset contained the `SparseCPU` key. Now, the `SparseCPU` dispatch key corresponds to two individual bits: the sparse bit and the cpu bit. So what you're really testing by writing that code is:
- does the keyset contain any sparse keys (is the `DispatchKey::Sparse` bit set)
- OR does the keyset contain any CPU backend keys (is the `BackendBit::CPUBit` bit set).

Reply:
We have the option to if we'd like to. The tradeoff of doing the same thing for `SparseCsr` is:

(+) We free up 1 extra dispatch key, since we can kill `SparseCsrCPU` + `SparseCsrCUDA`
(+) We give other backends the ability to add SparseCsr kernels without having to use up more DispatchKey bits
(-) We add another ~12 slots to every op's runtime dispatch table

Since `SparseCsr` and `Autocast` are both currently only used by two backends, I didn't bother switching them to be "per-backend" functionalities for now.

Reply:
>    // DispatchKeySet as the iterator stores a pointer to the raw representation
   // of the DispatchKeySet.
+  // Note: When we encounter a per-backend functionality (e.g. Dense or Sparse),
+  // we will iterate through EVERY backend in the keyset, for that functionality.
+  // For example, if the next functionality key to iterate over is Autograd,
+  // and the backend bits in the keyset correspond to [BackendBit::CPUBit, DispatchKey::CUDABit],
+  // then the next two keys we return will be DispatchKey::AutogradCPU, DispatchKey::AutogradCUDA
+  // (CPU first because it has lower precedence than CUDA in DispatchKey.h).